src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -328,144 +328,582 @@ private void TrySetUnrecoverableError(SyntaxDiagnosticInfo error)
+                            width: closeBraceCount,
+                            ErrorCode.ERR_TooManyCloseBracesForRawString));
+                    }
+                }
+            }
+
+            private void HandleOpenBraceInContent(InterpolatedStringKind kind, int startingDollarSignCount, ArrayBuilder<Interpolation>? interpolations)
+            {
+                if (kind is InterpolatedStringKind.Normal or InterpolatedStringKind.Verbatim)
+                {
+                    HandleOpenBraceInNormalOrVerbatimContent(kind, interpolations);
+                }
+                else
+                {
+                    HandleOpenBraceInRawContent(kind, startingDollarSignCount, interpolations);
+                }
+            }
+
+            private void HandleOpenBraceInNormalOrVerbatimContent(InterpolatedStringKind kind, ArrayBuilder<Interpolation>? interpolations)
+            {
+                Debug.Assert(kind is InterpolatedStringKind.Normal or InterpolatedStringKind.Verbatim);
+                if (_lexer.TextWindow.PeekChar(1) == '{')
+                {
+                    _lexer.TextWindow.AdvanceChar(2); // {{
+                }
+                else
+                {
+                    int openBracePosition = _lexer.TextWindow.Position;
+                    _lexer.TextWindow.AdvanceChar();
+                    ScanInterpolatedStringLiteralHoleBalancedText(kind, '}', isHole: true, out var colonRange);
+                    int closeBracePosition = _lexer.TextWindow.Position;
+                    if (_lexer.TextWindow.PeekChar() == '}')
+                    {
+                        _lexer.TextWindow.AdvanceChar();
+                    }
+                    else
+                    {
+                        TrySetError(_lexer.MakeError(openBracePosition - 1, 2, ErrorCode.ERR_UnclosedExpressionHole));
+                    }
+
+                    interpolations?.Add(new Interpolation(
+                        new Range(openBracePosition, openBracePosition + 1),
+                        colonRange,
+                        new Range(closeBracePosition, _lexer.TextWindow.Position)));
+                }
+            }
+
+            private void HandleOpenBraceInRawContent(InterpolatedStringKind kind, int startingDollarSignCount, ArrayBuilder<Interpolation>? interpolations)
+            {
+                Debug.Assert(kind is InterpolatedStringKind.SingleLineRaw or InterpolatedStringKind.MultiLineRaw);
+
+                // In raw content we are allowed to see up to 2*N-1 open (or close) curlies.  For example, if the string
+                // literal starts with `$$$"""` then we can see up to `2*3-1 = 5` curlies like so `$$$""" {{{{{`.  The
+                // inner three curlies start the interpolation.  The outer two curies are just content.  This ensures
+                // the rule that the content cannot contain a sequence of open or close curlies equal to (or longer)
+                // than the dollar sequence.
```

Comment:
>than [](http://example.com/codeflow?start=19&length=4)

typo: the "than" should probably inside the parens: `(or longer than) ...`

Reply:
done.